lib/remap-lang/src/diagnostic.rs

```diff
@@ -0,0 +1,203 @@
+                                Label::primary(LabelMessage::GotKind(*got), span.into()),
+                                Label::context(LabelMessage::ExpectedKind(*want), span.into()),
+                            ],
+                            notes: vec![Note::CoerceValue],
+                        }
+                    }
+                    _err => {}
+                },
+            },
+        };
+
+        Diagnostic {
+            severity: Severity::Error,
+            message: Message::Parse,
+            labels: vec![],
+            notes: vec![],
+        }
+    }
+}
+
+// -----------------------------------------------------------------------------
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Severity {
+    Bug,
+    Error,
+    Warning,
+    Note,
+}
+
+impl Into<diagnostic::Severity> for Severity {
+    fn into(self) -> diagnostic::Severity {
+        use Severity::*;
+
+        match self {
+            Bug => diagnostic::Severity::Bug,
+            Error => diagnostic::Severity::Error,
+            Warning => diagnostic::Severity::Warning,
+            Note => diagnostic::Severity::Note,
+        }
+    }
+}
+
+// -----------------------------------------------------------------------------
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Message {
+    Parse,
+    Fallible,
+    ReturnValue,
+    IfConditionType,
+}
+
+impl fmt::Display for Message {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use Message::*;
+
+        match self {
+            Parse => f.write_str("parse error"),
+            Fallible => f.write_str("uncaught error"),
+            ReturnValue => f.write_str("unexpected return value"),
+            IfConditionType => f.write_str("invalid if-condition type"),
+        }
+    }
+}
+
+// -----------------------------------------------------------------------------
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum LabelMessage {
+    GotKind(Kind),
+    ExpectedKind(Kind),
+}
+
+impl fmt::Display for LabelMessage {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use LabelMessage::*;
+
+        match self {
+            GotKind(kind) => write!(f, "got: {}", kind),
+            ExpectedKind(kind) => write!(f, "expected: {}", kind),
+        }
+    }
+}
+
+// -----------------------------------------------------------------------------
+
+#[derive(Debug, PartialEq, Clone, Copy)]
+pub enum Note {
+    ExpectedKind(Kind),
+    CoerceValue,
+
+    #[doc(hidden)]
+    SeeLangDocs,
+}
+
+impl fmt::Display for Note {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        use Note::*;
+
+        match self {
+            ExpectedKind(kind) => write!(f, "expected: {}", kind),
+            CoerceValue => {
+                f.write_str("hint: coerce the value using one of the coercion functions")
+            }
+            SeeLangDocs => f.write_str("see language documentation at: https://vector.dev"),
```

Comment:
```suggestion
            SeeLangDocs => f.write_str("see language documentation at: /docs/reference/remap-language/"),
```

Reply:
Thanks! I was planning to get the correct links (including to function documentations) before I merge this 👍